src/types/block-suggestion/index.ts

```diff
@@ -0,0 +1,109 @@
+import { Option } from '@slack/types';
+import { StringIndexed, XOR } from '../helpers';
+import { AckFn } from '../utilities';
+import { ViewOutput } from '../view/index';
+import { FunctionExecutionContext } from '../functions';
+
+/**
+ * Arguments which listeners and middleware receive to process a Block Suggestions request from Slack
+ */
+export interface SlackBlockSuggestionsMiddlewareArgs<Source extends BlockSuggestionsSource = BlockSuggestionsSource> {
+  payload: BlockSuggestionsPayloadFromType<Source>;
+  body: this['payload'];
+  options: this['payload'];
+  ack: BlockSuggestionsAckFn;
+}
+
+/**
+ * The source that comprises of Block Suggestions Requests.
+ */
+export type BlockSuggestionsSource = 'block_suggestion';
+
+export type SlackBlockSuggestion = BlockSuggestionPayload;
+
+export interface BasicBlockSuggestionsPayload<Type extends string = string> {
+  type: Type;
+  value: string;
+}
+
+export type BlockSuggestionsPayloadFromType<T extends string> = KnownBlockSuggestionsPayloadFromType<T> extends never
+  ? BasicBlockSuggestionsPayload<T>
+  : KnownBlockSuggestionsPayloadFromType<T>;
+
+export type KnownBlockSuggestionsPayloadFromType<T extends string> = Extract<SlackBlockSuggestion, { type: T }>;
+
+/**
+ * Block Suggestion payload model for next-gen interactivity
+ */
+export interface BlockSuggestionPayload extends StringIndexed {
```

Comment:
I took the payload from Fil's PR and had to tweak it (make some params optional and add more params) a bit based on what I observed for `block_suggestions` payloads in both enterprise and regular workspaces. If anyone is testing this, would love confirmation that the payloads look similar! You can view example shapes of payloads I was getting in the corresponding test file, `src/types/block-suggestion/index.spec.ts`